naresh/src/main/java/com/lftechnology/phpjava/ems/services/EmployeeService.java

```diff
@@ -0,0 +1,139 @@
+package com.lftechnology.phpjava.ems.services;
+
+import com.lftechnology.phpjava.ems.constants.Constant;
+import com.lftechnology.phpjava.ems.dao.EmployeeDaoImpl;
+import com.lftechnology.phpjava.ems.dao.UserDaoImpl;
+import com.lftechnology.phpjava.ems.entities.Employee;
+import com.lftechnology.phpjava.ems.entities.User;
+import com.lftechnology.phpjava.ems.enums.Role;
+import com.lftechnology.phpjava.ems.utlis.PasswordHashGenerator;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * EmployeeService
+ *
+ * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+ * @since August, 09 2016
+ */
+public class EmployeeService {
+
+    /**
+     * check if admin user exists or not and create a new one if it doesn't
+     *
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    public void checkAndCreateAdminUser() throws Exception {
+        if (!this.checkAdminUserExists()) {
+            Employee employee = new Employee();
+            employee.setFullname("Admin User")
+                    .setAddress("Somewhere")
+                    .setRole(Role.ADMIN)
+                    .setDepartment("Admin")
+                    .setUsername("admin")
+                    .setTerminated(false);
+            employee.setPassword(PasswordHashGenerator.getSaltedHash("password"));
+            insertUserAndEmployee(employee);
+
+        }
+    }
+
+    /**
+     * check if admin user exists or not
+     *
+     * @return boolean
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    public boolean checkAdminUserExists() throws SQLException {
+        EmployeeDaoImpl employeeDao = new EmployeeDaoImpl();
+        List<Employee> employees = employeeDao.findByRole(Role.ADMIN);
+        if (employees.isEmpty()) {
+            return false;
+        }
+        Employee employee = employees.get(0);
+        Boolean result = (employee.getRole() != null) ? employee.getRole().equals(Role.ADMIN) ? true : false : false;
+        return result;
+    }
+
+
+    public int addEmployee(Employee employee) throws Exception {
+        employee.setFullname(employee.getFullname().trim());
```

Comment:
generally you do not call .trim() method. correct object should come from the controller itself
.